src/timer.c

```diff
@@ -268,6 +272,13 @@ static int gettime(int casep)
 /*	Implements the SETTIME subr call, sub-dispatched from subr.c	*/
 /*									*/
 /************************************************************************/
+#ifdef MAIKO_OS_HAIKU
+int settimeofday(struct timeval *tv, struct timezone *tz)
+{
+  set_real_time_clock(tv->tv_sec);
```

Comment:
If `timezone` is ignored, it would probably be best for this method to be a no-op; just have it return `-1` and set `errno` to `EOPNOTSUPP`.

Reply:
@waddlesplash it only ever passes `NULL` to the `settimeofday()` call for the timezone.  The call-site doesn't care about the success/failure or any return value - it's a best-effort and isn't really expected to work on Unix-y systems, it really only mattered when Lisp *was* the operating system and then it could do the right thing (and that wasn't via settimeofday syscalls!)

Reply:
Well, if it isn't really expected to work, this really should be reduced to a no-op. `set_real_time_clock` is certainly not a no-op on Haiku!